state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -973,3 +1063,67 @@ impl<T: Config> OnKilledAccount<T::AccountId> for Pallet<T> {
 		BoundRedeemAddress::<T>::remove(account_id);
 	}
 }
+
+#[derive(Clone, PartialEq, Eq, Encode, Decode, TypeInfo, DebugNoBound)]
+#[scale_info(skip_type_params(T))]
+pub enum AllowedCallsViaSCGateway<T: Config> {
+	Delegate {
+		delegator: EthereumAddress, // Ethereum Address of the delegator
+		operator: T::AccountId,     // Operator the amount to delegate to
+	},
+	Undelegate {
+		delegator: EthereumAddress, // Ethereum Address of the delegator
+		operator: T::AccountId,     // Operator the amount was delegated to
+	},
+}
+
+// calls via vault and transfers for future use
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, TypeInfo, DebugNoBound)]
+pub enum AllowedCallsViaVault {}
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, TypeInfo, DebugNoBound)]
+pub enum AllowedCallsViaTransfer {}
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, TypeInfo, DebugNoBound)]
+pub enum AllowedCallsOnlyCalls {}
```

Comment:
I'm not sure these really need to be separate enums. Either way if they are unused we don't need to add them.